markdownextradata/plugin.py

```diff
@@ -16,6 +21,41 @@ class MarkdownExtraDataPlugin(BasePlugin):
     """
     Inject certain config variables into the markdown
     """
+
+    config_scheme = (
+        ('data', mkdocs.config.config_options.Type(mkdocs.utils.string_types, default=None)),
+    )
+
+    def __add_data__(self, config, namespace, data):
+        # creates the namespace and adds the data there
+        namespace = ['extra'] + namespace.split(os.sep)
+        holder = config
+        while len(namespace) > 1:
+            if not namespace[0] in holder: holder[namespace[0]] = {}
+            holder = holder[namespace[0]]
+            del namespace[0]
+        holder[namespace[0]] = data
+
+    def on_pre_build(self, config):
+        # this loads all data from the supplied data directory, or otherwise a _data directory next to mkdocs.yml or inside the docs_dir. Does nothing if the dir does not exist.
+
+        data = self.config.get('data')
+        for datadir in [ os.path.dirname(config['config_file_path']), config['docs_dir'] ]:
+            if not data:
+                data = os.path.join(datadir, '_data')
+                if not os.path.exists(data): data = None
+
+        if data and os.path.exists(data):
+            for filename in Path(data).glob('**/*.yml'):
```

Comment:
instead of 2 for loops rather something like `Path(data).glob('**/*.yml') + Path(data).glob('**/*.json')` or maybe even `Path(data).glob('**/*.yml|*.json')` cant remember of glob supports this or not (sorry on holiday in the sun atm)

Reply:
pathlob globs don't support groups, but it's possible to chain them.